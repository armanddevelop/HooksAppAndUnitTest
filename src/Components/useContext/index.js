export { HomeScreen } from "./HomeScreen";
export { AboutScreen } from "./AboutScreen";
export { LognScreen } from "./LognScreen";
export { NavBar } from "./NavBar";
