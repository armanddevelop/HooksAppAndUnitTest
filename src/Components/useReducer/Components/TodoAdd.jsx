import React from "react";
import { useForm } from "../../../Hooks/useForm";
import PropTypes from "prop-types";
export const TodoAdd = ({ handleAddToDo }) => {
  const { formState, handleInputChange, resetField } = useForm({
    description: "",
  });
  const { description } = formState;
  const handleSubmit = (e) => {
    e.preventDefault();
    if (description.trim().length === 0) return;
    const newTodo = {
      id: new Date().getTime(),
      description,
      done: false,
    };
    handleAddToDo(newTodo);
    resetField();
  };
  return (
    <>
      <h4>Add Task</h4>
      <hr />
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          name="description"
          placeholder="learn code"
          autoComplete="off"
          className="form-control"
          value={description}
          onChange={handleInputChange}
        ></input>
        <button
          type="submit"
          className="btn btn-outline-primary mt-1 btn-block"
        >
          ADD
        </button>
      </form>
    </>
  );
};
TodoAdd.propTypes = {
  handleAddToDo: PropTypes.func.isRequired,
};
