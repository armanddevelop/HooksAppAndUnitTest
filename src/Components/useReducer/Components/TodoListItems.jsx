import PropTypes from "prop-types";
export const TodoListItems = ({
  description,
  done,
  idx,
  id,
  handleDelete,
  handleToogle,
}) => {
  return (
    <li className="list-group-item">
      <p
        className={`${done ? "complete" : ""}`}
        onClick={() => handleToogle(id)}
      >
        {idx + 1}.- {description}
      </p>
      <button onClick={() => handleDelete(id)} className="btn btn-danger">
        Delete
      </button>
    </li>
  );
};
TodoListItems.prototype = {
  description: PropTypes.string.isRequired,
  done: PropTypes.bool.isRequired,
  id: PropTypes.number.isRequired,
  idx: PropTypes.number.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToogle: PropTypes.func.isRequired,
};
