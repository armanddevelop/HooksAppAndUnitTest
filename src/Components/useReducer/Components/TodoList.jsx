import { TodoListItems } from "./TodoListItems";
import PropTypes from "prop-types";

export const TodoList = ({ todos, handleDelete, handleToggle }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map(({ id, description, done }, idx) => {
        return (
          <TodoListItems
            key={id}
            description={description}
            done={done}
            idx={idx}
            id={id}
            handleDelete={handleDelete}
            handleToogle={handleToggle}
          />
        );
      })}
    </ul>
  );
};
TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToggle: PropTypes.func.isRequired,
};
