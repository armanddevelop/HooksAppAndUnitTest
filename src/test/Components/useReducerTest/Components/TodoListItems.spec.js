import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import toJson from "enzyme-to-json";
import { TodoListItems } from "../../../../Components/useReducer/Components/TodoListItems";

describe("Test for TodoListItems", () => {
  const handleDelete = jest.fn();
  const handleToogle = jest.fn();
  const container = shallow(
    <TodoListItems
      description="aprender Docker"
      done={false}
      idx={0}
      id={1}
      handleDelete={handleDelete}
      handleToogle={handleToogle}
    />
  );

  it("Take Snapshot to todoListItems", () => {
    expect(toJson(container)).toMatchSnapshot();
  });

  it("Should call handleDelete", () => {
    container.find("button").simulate("click", () => {
      handleDelete();
    });
    expect(handleDelete).toHaveBeenCalledWith(1);
  });

  it("Should call handleToogle", () => {
    container.find("p").simulate("click", () => {
      handleToogle();
    });
    expect(handleToogle).toHaveBeenCalledWith(1);
  });

  it("Should show className empty", () => {
    const text = container.find("p").text();
    const completeClass = container.find("p").hasClass("complete");
    expect(container.find("p").text()).toBe(text);
    expect(completeClass).toBeFalsy();
  });

  it("Should show className complete", () => {
    const container = shallow(
      <TodoListItems
        description="aprender Docker"
        done={true}
        idx={0}
        id={1}
        handleDelete={handleDelete}
        handleToogle={handleToogle}
      />
    );
    const completeClass = container.find("p").hasClass("complete");
    expect(completeClass).toBeTruthy();
  });
});
