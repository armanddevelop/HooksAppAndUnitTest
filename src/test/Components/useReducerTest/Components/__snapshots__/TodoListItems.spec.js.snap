// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for TodoListItems Take Snapshot to todoListItems 1`] = `
<li
  className="list-group-item"
>
  <p
    className=""
    onClick={[Function]}
  >
    1
    .- 
    aprender Docker
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
